prime-router/src/main/kotlin/azure/DatabaseAccess.kt

```diff
@@ -484,6 +504,55 @@ class DatabaseAccess(private val create: DSLContext) : Logging {
             ?.getValue(DSL.max(SETTING.VERSION)) ?: -1
     }
 
+    fun saveTestData(testData: List<TestData>, txn: DataAccessTransaction) {
+        testData.forEach {
+            DSL
+                .using(txn)
+                .insertInto(TEST_DATA)
+                .set(TEST_DATA.TRACKING_ID, it.trackingId)
+                .set(TEST_DATA.REPORT_ID, it.reportId)
+                .set(TEST_DATA.ORDERING_PROVIDER_NAME, it.orderingProviderName)
+                .set(TEST_DATA.ORDERING_PROVIDER_ID, it.orderingProviderId)
+                .set(TEST_DATA.ORDERING_PROVIDER_STATE, it.orderingProviderState)
+                .set(TEST_DATA.ORDERING_PROVIDER_POSTAL_CODE, it.orderingProviderPostalCode)
+                .set(TEST_DATA.ORDERING_PROVIDER_COUNTY, it.orderingProviderCounty)
+                .set(TEST_DATA.ORDERING_FACILITY_COUNTY, it.orderingFacilityCounty)
+                .set(TEST_DATA.TEST_RESULT_CODE, it.testResultCode)
+                .set(TEST_DATA.TEST_RESULT, it.testResult)
+                .set(TEST_DATA.EQUIPMENT_MODEL, it.equipmentModel)
+                .set(TEST_DATA.ORDERING_FACILITY_CITY, it.orderingFacilityCity)
+                .set(TEST_DATA.ORDERING_FACILITY_COUNTY, it.orderingFacilityCounty)
+                .set(TEST_DATA.ORDERING_FACILITY_NAME, it.orderingFacilityName)
+                .set(TEST_DATA.ORDERING_FACILITY_POSTAL_CODE, it.orderingFacilityPostalCode)
+                .set(TEST_DATA.ORDERING_FACILITY_STATE, it.orderingFacilityState)
+                .set(TEST_DATA.TESTING_LAB_CITY, it.testingLabCity)
+                .set(TEST_DATA.TESTING_LAB_CLIA, it.testingLabClia)
+                .set(TEST_DATA.TESTING_LAB_COUNTY, it.testingLabCounty)
+                .set(TEST_DATA.TESTING_LAB_NAME, it.testingLabName)
+                .set(TEST_DATA.TESTING_LAB_STATE, it.testingLabState)
+                .set(TEST_DATA.TESTING_LAB_POSTAL_CODE, it.testingLabPostalCode)
+                .set(TEST_DATA.PATIENT_COUNTY, it.patientCounty)
+                .set(TEST_DATA.PATIENT_ETHNICITY_CODE, it.patientEthnicityCode)
+                .set(TEST_DATA.PATIENT_ETHNICITY, it.patientEthnicity)
+                .set(TEST_DATA.PATIENT_GENDER_CODE, it.patientGenderCode)
+                .set(TEST_DATA.PATIENT_GENDER, it.patientGender)
+                .set(TEST_DATA.PATIENT_POSTAL_CODE, it.patientPostalCode)
+                .set(TEST_DATA.PATIENT_RACE_CODE, it.patientRaceCode)
+                .set(TEST_DATA.PATIENT_RACE, it.patientRace)
+                .set(TEST_DATA.PATIENT_STATE, it.patientState)
+                .set(TEST_DATA.PATIENT_AGE, it.patientAge)
+                .set(TEST_DATA.SPECIMEN_COLLECTION_DATE_TIME, it.specimenCollectionDateTime)
+                .executeAsync()
+        }
+    }
```

Comment:
Set this up to execute async so it can just run in the background and we don't have to monitor it. I think it will be fine in production, but we will have to see if it continues to hold up and perform well.